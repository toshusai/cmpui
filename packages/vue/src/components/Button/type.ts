export const ButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type ButtonVariant = (typeof ButtonVariant)[keyof typeof ButtonVariant];

export const ButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type ButtonSize = (typeof ButtonSize)[keyof typeof ButtonSize];

export type ButtonProps = {
  variant: ButtonVariant;
  size: ButtonSize;
  disabled: boolean;
};
