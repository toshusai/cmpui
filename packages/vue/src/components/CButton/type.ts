export const CButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type CButtonVariant =
  (typeof CButtonVariant)[keyof typeof CButtonVariant];

export const CButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type CButtonSize = (typeof CButtonSize)[keyof typeof CButtonSize];

export type CButtonProps = {
  variant?: CButtonVariant;
  size?: CButtonSize;
  disabled?: boolean;
};
