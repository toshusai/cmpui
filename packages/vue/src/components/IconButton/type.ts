export const IconButtonSize = {
  S: "S",
  M: "M",
} as const;

export type IconButtonSize =
  (typeof IconButtonSize)[keyof typeof IconButtonSize];

export type IconButtonProps = {
  bordered?: boolean;
  disabled?: boolean;
  rounded?: boolean;
  selected?: boolean;
  size?: IconButtonSize;
};
