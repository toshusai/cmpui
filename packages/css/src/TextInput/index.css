.cmpui_text-input__root {
  background-color: var(--cmpui-bg-color);
  border: 1px solid var(--cmpui-border-color);

  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  display: flex;
  height: 32px;
  justify-content: flex-start;
  position: relative;
  text-decoration: none;
}

.cmpui_text-input__root:focus-within {
  outline: var(--cmpui-primary-color) solid 1px;
  outline-offset: 1px;
}

.cmpui_text-input__root[aria-disabled="true"] {
  cursor: default;
  opacity: 0.5;
}

.cmpui_text-input__root[aria-invalid="true"] {
  border: 1px solid var(--cmpui-error-color);
}
.cmpui_text-input__root[aria-invalid="true"]:hover {
  border: 1px solid var(--cmpui-error-color);
}

.cmpui_text-input__root:hover {
  border: 1px solid var(--cmpui-border-hover-color);
}

.cmpui_text-input__root[aria-disabled="true"]:hover {
  border: 1px solid var(--cmpui-border-color);
}

.cmpui_text-input__label {
  background-image: linear-gradient(
    to bottom,
    transparent 10%,
    var(--cmpui-bg-color) 40%,
    var(--cmpui-bg-color) 60%,
    transparent 60%
  );
  color: var(--cmpui-text2-color);

  font-size: 12px;
  line-height: 14px;
  padding: 0 4px;
  position: absolute;
  transform: translate(4px, -16px);
}
.cmpui_text-input__root[aria-invalid="true"] > .cmpui_text-input__label {
  color: var(--cmpui-error-color);
}

.cmpui_text-input__input:nth-child(3) {
  padding-left: 0px;
}

.cmpui_text-input__input {
  color: var(--cmpui-text-color);
  font-family: var(--cmpui-font-family);

  background: transparent;
  border: none;
  box-sizing: border-box;
  display: flex;
  font-size: 16px;
  height: 100%;
  line-height: 16px;
  outline: none;
  padding: 0 8px;
  width: 100%;
}

.cmpui_text-input__input::placeholder {
  color: var(--cmpui-text3-color);
  font-style: italic;
}
