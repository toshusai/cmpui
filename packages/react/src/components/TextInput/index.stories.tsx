import { useState } from "react";
import type { Meta, StoryObj } from "@storybook/react";
import { IconCurrencyDollar, IconEye, IconEyeOff } from "@tabler/icons-react";

import { IconButton } from "../IconButton";

import { TextInput } from ".";

const meta: Meta<typeof TextInput> = {
  title: "Form/TextInput",
  component: TextInput,
  args: {
    label: "Label",
  },
};

export default meta;
type Story = StoryObj<typeof TextInput>;

export const Basic: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        placeholder="Placeholder"
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
    );
  },
};

export const Label: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        id="text-input-label"
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
    );
  },
};

export const Placeholder: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label=""
        placeholder="Placeholder"
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
    );
  },
};

export const Prefix: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Amount"
        placeholder="1.0"
        type="number"
        value={value}
        onChange={(event) => setValue(event.target.value)}
        prefix={
          <IconCurrencyDollar
            size={16}
            style={{ margin: "0 4px", minWidth: "16px" }}
          />
        }
      />
    );
  },
};

export const Suffix: Story = {
  render: function Render() {
    const [showPassword, setShowPassword] = useState(false);
    const [value, setValue] = useState("NN2mdJec,Ff+^[a");
    return (
      <TextInput
        label="Password"
        placeholder="password"
        type={showPassword ? "text" : "password"}
        suffix={
          <IconButton
            onClick={() => setShowPassword((v) => !v)}
            size="S"
            style={{ marginRight: "4px" }}
          >
            {showPassword ? (
              <IconEyeOff
                size={16}
                style={{ margin: "0 4px", minWidth: "16px" }}
              />
            ) : (
              <IconEye
                size={16}
                style={{ margin: "0 4px", minWidth: "16px" }}
              />
            )}
          </IconButton>
        }
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
    );
  },
};

export const Disabled: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        value={value}
        onChange={(event) => setValue(event.target.value)}
        disabled
      />
    );
  },
};

export const Invalid: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        value={value}
        onChange={(event) => setValue(event.target.value)}
        invalid
      />
    );
  },
};
