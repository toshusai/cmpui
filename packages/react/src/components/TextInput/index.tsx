import { forwardRef, memo, useRef } from "react";

import { mergeRefs } from "../..";
import { classNames } from "../../utils/classNames";
import { useNativeOnChange } from "../../utils/interactions/useNativeOnChange";

import "./index.css";

export type TextInputProps = {
  htmlPrefix?: string;
  invalid?: boolean;
  label?: React.ReactNode;
  placeholder?: string;
  prefix?: React.ReactNode;
  suffix?: React.ReactNode;
  value?: string;
  onNativeChange?: (e: Event) => void;
} & Omit<React.InputHTMLAttributes<HTMLInputElement>, "prefix">;

export const TextInput = memo(
  forwardRef<HTMLInputElement, TextInputProps>(function TextInput(
    {
      className,
      htmlPrefix,
      invalid,
      label,
      placeholder,
      prefix,
      suffix,
      value,
      onNativeChange,
      ...props
    },
    fRef,
  ) {
    const ref = useRef<HTMLInputElement>(null);
    useNativeOnChange(ref, onNativeChange);
    return (
      <div
        className={classNames("cmpui_text-input__root", className)}
        aria-disabled={props.disabled}
        aria-invalid={invalid}
      >
        {prefix}
        {typeof label === "string" ? (
          <label className="cmpui_text-input__label" htmlFor={props.id}>
            {label}
          </label>
        ) : (
          label
        )}
        <input
          ref={mergeRefs(ref, fRef)}
          type="text"
          className="cmpui_text-input__input"
          value={value ?? ""}
          placeholder={placeholder ?? ""}
          prefix={htmlPrefix}
          aria-invalid={invalid}
          {...props}
        />
        {suffix}
      </div>
    );
  }),
);
