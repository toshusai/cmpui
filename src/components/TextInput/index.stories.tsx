import type { Meta, StoryObj } from "@storybook/react";
import { useState } from "react";
import { TextInput } from ".";

const meta: Meta<typeof TextInput> = {
  title: "Components/Form/TextInput",
  component: TextInput,
  args: {
    label: "Label",
  },
};

export default meta;
type Story = StoryObj<typeof TextInput>;

export const Basic: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        placeholder="Placeholder"
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
    );
  },
};

export const Label: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        id="text-input-label"
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
    );
  },
};

export const Placeholder: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label=""
        placeholder="Placeholder"
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
    );
  },
};

export const Suffix: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        prefix={
          <svg
            xmlns="http://www.w3.org/2000/svg"
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            style={{
              maxWidth: 16,
              height: 16,
            }}
          >
            <title>search</title>
            <path d="M784-120 532-372q-30 24-69 38t-83 14q-109 0-184.5-75.5T120-580q0-109 75.5-184.5T380-840q109 0 184.5 75.5T640-580q0 44-14 83t-38 69l252 252-56 56ZM380-400q75 0 127.5-52.5T560-580q0-75-52.5-127.5T380-760q-75 0-127.5 52.5T200-580q0 75 52.5 127.5T380-400Z" />
          </svg>
        }
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
    );
  },
};

export const Disabled: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        value={value}
        onChange={(event) => setValue(event.target.value)}
        disabled
      />
    );
  },
};

export const Invalid: Story = {
  render: function Render() {
    const [value, setValue] = useState("");
    return (
      <TextInput
        label="Label"
        value={value}
        onChange={(event) => setValue(event.target.value)}
        invalid
      />
    );
  },
};
