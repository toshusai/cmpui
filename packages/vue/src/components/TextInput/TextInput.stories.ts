import type { Meta, StoryObj } from "@storybook/vue3";
import TextInput from "./TextInput.vue";

const meta = {
  title: "inputs/TextInput",
  component: TextInput,
} satisfies Meta<typeof TextInput>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    default: "Label",
    placeholder: "Placeholder",
  },
};
