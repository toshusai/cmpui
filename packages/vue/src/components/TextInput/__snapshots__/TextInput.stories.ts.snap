// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/TextInput Default smoke-test 1`] = `
<div class="cmpui_text-input__root"
     aria-disabled="false"
     aria-invalid="false"
     default="Label"
     placeholder="Placeholder"
>
  <label class="cmpui_text-input__label"
         for="id-1"
  >
    Label
  </label>
  <input id="id-1"
         default="Label"
         placeholder="Placeholder"
         aria-invalid="false"
         class="cmpui_text-input__input"
         type="text"
  >
</div>
`;
